### PR TITLE
(PC-17481) feat(search): add type filter button search results

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-17481

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|Capture d’écran 2022-09-23 à 16 35 38|         |Capture d’écran 2022-09-23 à 16 35 20|Capture d’écran 2022-09-23 à 16 34 48|                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/search/components/SearchResults.tsx [hunk 1]
```diff
@@ -79,6 +82,12 @@ export const SearchResults: React.FC = () => {
   const maxPrice: number | undefined = getPriceAsNumber(params?.maxPrice)
   const priceIsEntered = (minPrice !== undefined && minPrice > 0) || maxPrice !== undefined
 
+  const typeIsEntered =
```
Comment (kopax-polyconseil) [inline src/features/search/components/SearchResults.tsx:85]: `hasType`, c'est une convention largement utilisé pour ton use case.

File: src/features/search/components/__tests__/SearchResults.native.test.tsx [hunk 2]
```diff
@@ -170,11 +170,57 @@ describe('SearchResults component', () => {
...
+    })
+
+    it.each`
+      type               | params
+      ${'duo offer'}     | ${{ offerIsDuo: true }}
+      ${'digital offer'} | ${{ offerTypes: { isDigital: true, isEvent: false, isThing: false } }}
+      ${'event offer'}   | ${{ offerTypes: { isDigital: false, isEvent: true, isThing: false } }}
+      ${'thing offer'}   | ${{ offerTypes: { isDigital: false, isEvent: false, isThing: true } }}
+    `(
+      'should display an icon and change color in type button when $type selected',
```
Comment (kopax-polyconseil) [inline src/features/search/components/__tests__/SearchResults.native.test.tsx:196]: `should display an icon and change color in type button when has $type selected`

(ajout de has)

File: src/features/search/components/SearchResults.tsx [hunk 3]
```diff
@@ -61,6 +62,8 @@ export const SearchResults: React.FC = () => {
   const { navigate } = useNavigation<UseNavigationType>()
   const { section } = useLocationType(searchState)
   const { label: locationLabel } = useLocationChoice(section)
+  const { data: appSettings } = useAppSettings()
+  const filtersButtonsDisplay = appSettings?.appEnableCategoryFilterPage ?? false
```
Comment (voisinhugo) [inline src/features/search/components/SearchResults.tsx:66]: `shouldDisplayFiltersButtons` plutôt ? Ou n'importe quel nom qui garde l'idée que c'est un booléen :)

Comment (kopax-polyconseil) [reply src/features/search/components/SearchResults.tsx:66]: Oui, `should|has|is` sont des suffixes courant pour les booleans, prend juste une notation standard, https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

**Review by voisinhugo — APPROVED** [review]

**Review by kopax-polyconseil — APPROVED** [review]

**Pull request merged**